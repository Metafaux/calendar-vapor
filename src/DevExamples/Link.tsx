import React, { useState } from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

interface LinkProps {
  page: string;
  children: React.ReactNode;
}

const Link = ({ page, children }: LinkProps) => {
  const [classValue, setClassValue] = useState({ class: STATUS.NORMAL });

  const onMouseEnter = () => {
    setClassValue({ class: STATUS.HOVERED });
  };

  const onMouseLeave = () => {
    setClassValue({ class: STATUS.NORMAL });
  };

  return (
    <a
      className={classValue.class}
      href={page}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

export default Link;
