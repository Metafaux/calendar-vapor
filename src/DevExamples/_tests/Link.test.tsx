import React from 'react';
import renderer from 'react-test-renderer';
import Link from '../Link';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <Link page="https://www.metafaux.com">Metafaux</Link>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  /* @ts-ignore */
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  /* @ts-ignore */
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
