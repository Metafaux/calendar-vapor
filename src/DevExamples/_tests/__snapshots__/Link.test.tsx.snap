// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://www.metafaux.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Metafaux
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://www.metafaux.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Metafaux
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://www.metafaux.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Metafaux
</a>
`;
